commons/all/src/main/java/org/infinispan/commons/hash/CRC16.java

```diff
@@ -35,9 +35,9 @@ public static CRC16 getInstance() {
    public int hash(byte[] payload) {
       int crc = 0x0000;
       for (byte b : payload) {
-         crc = (crc >>> 8) ^ CRC16Table.table[(crc ^ b) & 0xff];
+         crc = ((crc << 8) ^ CRC16Table.table[((crc >>> 8) ^ (b & 0xFF)) & 0xFF]);
       }
-      return crc;
+      return crc & 0xFFFF;
```

Comment:
Good catch, was this generating hashes outside of the 16 bit range before?

Reply:
Found it when comparing with the results from Lettuce. It was returning all weird values.